projects/libs/core-library-typescript/src/virtual-machine/hard-coded-contracts/registry/kin-smart-contract.ts

```diff
@@ -0,0 +1,70 @@
+import BaseSmartContract from "../base-smart-contact";
+
+export default class KinAtnSmartContract extends BaseSmartContract {
+  static readonly ALLOCATED_ACCOUNT_NAME = "repository_source";
```

Comment:
I wouldn't use an account for initial token allocation, but rather just a counter. Otherwise you start running into issues such as how to prevent someone from transferring tokens back to the allocated pool, and the fact that the account name doesn't meet the addressing scheme format

Reply:
what do you mean by counter? i'll take this on slack